src/OrchardCore.Modules/OrchardCore.Forms/FormShapeTableProvider.cs

```diff
@@ -0,0 +1,27 @@
+using OrchardCore.ContentManagement;
+using OrchardCore.DisplayManagement;
+using OrchardCore.DisplayManagement.Descriptors;
+using OrchardCore.DisplayManagement.Utilities;
+using OrchardCore.Forms.Models;
+
+namespace OrchardCore.Forms;
+
+public class FormShapeTableProvider : IShapeTableProvider
+{
+    public void Discover(ShapeTableBuilder builder)
+    {
+        builder.Describe("Widget__Form")
+            .OnDisplaying(context =>
+            {
+                if (context.Shape.TryGetProperty<ContentItem>("ContentItem", out var contentItem) && contentItem.ContentType == "Form")
+                {
+                    context.Shape.Metadata.Alternates.Add($"Widget__Form_{contentItem.ContentItemId}");
```

Comment:
Actually, how about adding this for all content? In `OrchardCore.Contents.Shapes` we have an alternate with the int `Id`, but that doesn't seem too useful.

Reply:
Not bad idea. but maybe in a separate PR. I can remove it from here.

Reply:
It's just that if you need this for Form Widgets, then you can achieve the same by adding pretty much ln 18 to the other class instead and it's done. I don't think it's worth getting back to this just for that (since it's a non-intrusive, non-breaking change). However, it needs to be documented under https://docs.orchardcore.net/en/latest/docs/reference/modules/Templates/#content-templates.

Reply:
OK then, and we can do this later: https://github.com/OrchardCMS/OrchardCore/issues/15572